doc/developer_guide/psyir_symbols.rst

```diff
@@ -429,6 +432,47 @@ instances:
 .. autoclass:: psyclone.core.access_info.AccessInfo
     :members:
 
+Indices
+-------
+The `AccessInfo` class stores the original PSyIR node that contains the
+access, but it also stores the indices used in a simplified form, which
+makes it easier to analyse dependencies without having
+to analyse a PSyIR tree for details. Using the `indices_groups` property of
+an `AccessInfo` object returns a list containing the indices used for
+each component. For example, an access like `a(i)%b(j,k)%c(l)` would
+return the list `[ [i], [j,k], [l] ]`. In case of non-array accesses,
+the corresponding index list will be empty, e.g. `a%b(j)%c` will
+have the indices groups `[ [], [j], [] ]`, and a scalar `a` will just
+return `[ [] ]`.
+
+Each entry in this list of lists is still a PSyIR node for each index
+expressions used. To find out details about an index expression, you can
+either analyse the tree, or use the variable access functionality again.
+Below an example that shows how this is done to determine if an array expression
+contains a reference to a given variable `index_variable`. The variable
+`access_info` is an instance of `AccessInfo` and contains the information
+about one reference. The outer loop is over the indices used for each component,
+i.e. in case of `a(i)%b(j,k)%c(l)` it would loop over the three
+members of the list `[ [i], [j,k], [l] ]`. The inner loop then
+tests each index used, e.g. it would be `[i]` for the first index group,
+then `[j, k]` etc. Then the function `reference_accesses` is used to
+analyse the index expression.
+
+.. code-block:: python
+
+  for index_group in access_info.indices:
```

Comment:
I'm not entirely comfortable here. If we had a Reference, `sref`, and wanted to check whether a variable named `index_variable` was accessed we could do:

    refs = sref.walk(Reference)
    for ref in refs:
        if ref.symbol.name == index_variable:
            return True

Am I missing something?